add callback argument to lumo get-completions

hlolli opened the pull request:
lumo.repl/get-completions takes two arguments.
https://github.com/anmonteiro/lumo/blob/master/src/cljs/snapshot/lumo/repl.cljs#L1252
So I added map str function into the callback, now I'm getting completions suggestions from company-mode.

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x ] The commits are consistent with our [contribution guidelines][1]
- [x ] The new code is not generating bytecode or `M-x checkdoc` warnings
- [ ] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

[1]: https://github.com/clojure-emacs/inf-clojure/blob/master/.github/CONTRIBUTING.md

hlolli commented:
Yes, the completions work, but arglists aren't appearing while scrolling the completion list via company. Could you test this and the other pull request I made, or look over it. I see that on lumo side, sometimes lumo gives wrong completions, there is to say, all possible completions, in my case after ) character.

arichiardi commented:
So the callback thing is a pain (notice the `nil` at the end because of the print) :

```
cljs.user=> (lumo.repl/get-completions "redu" #(print (map str %)))
(reduce reduce-kv reduceable? reduced reduced? reductions)nil
```

hlolli commented:
That nil should go trough the elisp `read` function and become ignored when it hit the case statement. I regret not making two pull requests as I see this are two different problems interleaved with one another.

arichiardi commented:
I'd solve it this way:

```
(let [hack (atom)]
  (lumo.repl/get-completions "redu" #(swap! hack (fn [r] (map str %))))
  @hack)
```

hlolli commented:
Very good! Agree, I update the pull request and use your snippet :smile:

arichiardi commented:
Fixed.

hlolli commented:
Are you commenting on the correct pull requests? This fixes it or you've fixed it?

arichiardi commented:
No sorry I have fixed the comment with the snippet, I cannot fix the PR :smile:

arichiardi commented:
This has another problem,I will raise it in `lumo`.

hlolli commented:
PR updated.

arichiardi commented:
As per the linked `lumo` issue, this avoid pretty pretting:

```
cljs.user=> (do (set! lumo.repl/*pprint-results* false) (let [hack (atom)] (lumo.repl/get-completions "redu" #(reset! hack (map str %))) @hack))
("Reduced" "reduce" "reduce-kv" "reduceable?" "reduced" "reduced?" "reductions")
```

arichiardi commented on inf-clojure.el:
```diff
@@ -726,7 +726,10 @@ If you are using REPL types, it will pickup the most approapriate
 
 (defcustom inf-clojure-completion-form-lumo
-  "(doall (map str (lumo.repl/get-completions \"%s\")))"
+  "(let [ret (atom)]
```
If you set it to:

`(do (set! lumo.repl/*pprint-results* false) (let [hack (atom)] (lumo.repl/get-completions "redu" #(reset! hack (map str %))) @hack))`

and squash all commits into one, we are good :smile:

hlolli replied:
I'll do that and squash. But isn't this a bit mean to someone who wants to prettyprint the results when emacs is always setting it to false?

arichiardi commented:
Uhm, Antonio suggested it but you are right, it is global...Maybe this is the moment when we introduce a post processing step to completion. First I would try if it works with multiline lists, if not we have a fn that makes everything single line but we need to introduce some glue for post processing..

hlolli commented:
There are functions that can do that, but as far as I see, it returns a list of strings which is read as emacs data, so newline should be irrelevant, in company code completions list it makes no difference at least.

hlolli commented:
Ok I changed `swap!` to `reset!` it works and gives completions, and also squashed the commits.

Little related @arichiardi , there's a bug in lumo
![lumo_infclojure](https://user-images.githubusercontent.com/6074754/27034766-3de33bbe-4f7f-11e7-821f-62204cd9f02d.png)
This can be reproduced in lumo like this
```
(let [ret (atom)] (lumo.repl/get-completions "red)" (fn [res] (reset! ret (map str res)))) @ret)
```
Otherwise I think @bbatsov can merge this now.

arichiardi commented:
LGTM
